packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2168,9 +2168,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     }
 
     public readonly summarizeOnDemand: ISummarizer["summarizeOnDemand"] = (...args) => {
-        return this.clientDetails.type === summarizerClientType
-            ? this.summarizer.summarizeOnDemand(...args)
-            : this.summaryManager.summarizeOnDemand(...args);
+        return this.summaryManager !== undefined
+            ? this.summaryManager.summarizeOnDemand(...args)
+            : this.summarizer.summarizeOnDemand(...args);
```

Comment:
Pointless/confusing code reduce?:
```suggestion
        return (this.summaryManager ?? this.summarizer).summarizeOnDemand(...args);
```

But in actually, this is slightly different behavior than before. The intention is to only call the `summarizer.summarizeOnDemand` on summarizer clients. We don't want to call it on Summarizers for non-summarizer clients (i.e. when it's disabled). Although both are "error" cases and will throw, I think it would be good to have a more accurate error stopped here.

Reply:
Made the check more explicit in latest, and throw an error if it's called with summaries disabled (side thought - should this method be provided on `ContainerRuntime` at all vs. just providing access to the appropriate `Summarizer` to whoever needs to call it?  I'm not sure who is calling it though).

Reply:
Mostly it will be used by tests for now, but in the future consumers of the runtime may choose to disable heuristics and run at their own discretion.